api-spec.md

```diff
@@ -40,6 +54,23 @@ Implementations may **optionally** provide support for the full superset of STAC
 where the collection ID in the path is equivalent to providing that single value in the `collections` query parameter in 
 STAC API.
 
+### STAC API Compliance
```

Comment:
How to expose only STAC Collections in an API?

Reply:
I'm open on what exactly we do here, was just trying to capture what was in the PR.

My thinking was that a Collection only endpoint is actually _not_ a STAC API. In my mind a STAC API is when you provide 'search'. OpenEO is a compliant STAC catalog, as is https://storage.googleapis.com/pdd-stac/disasters-0.9.0/catalog.json And I'd see implementations that are servers (RESTful API's) that only implement STAC core, and don't offer search/ or collections/items/ endpoints.

I think I agree with your point about RESTful, with no specific endpoints required, but we should be able to still specify functionality required in some way? Is OGC Features API RESTful?

Reply:
> OpenEO is a compliant STAC catalog, as is https://storage.googleapis.com/pdd-stac/disasters-0.9.0/catalog.json

Unfortunately, it is not. We use the data rel type instead of child as it is specified in STAC API. So it's actually not a valid STAC catalog... And then /collections is also not valid STA, it's only valid in STAC API.

> I think I agree with your point about RESTful, with no specific endpoints required, but we should be able to still specify functionality required in some way?

Why would that be required? If we have conformance classes anyway, couldn't we just specify what is implemented there or just follow links? I'm not sure myself yet, but I don't like that it's obviously getting towards just STAC API for Items and it seems I need to come up with my own STAC API spec for Collections.

Reply:
Ah, that's interesting on using data rel type instead of child, I'd thought they were child. I do think that's a confusing little area.

I'm not set on not having STAC API just for items, just saying that was more of my mental model. That everyone should implement STAC Core, and that you layer on Features API / Search endpoint for filtering items. And the latter is 'STAC API'. 

But for beta.1 I prefer to not rock the boat - mostly just want to align with CQL / Transactions, clean things up, and get something for people to work with. I definitely see a beta.2, and I think the recommendations for how to implement STAC catalogs / collections is core to figure out, and this to me fits in. I think the conformance classes are probably the way to go. And I'm not set on my mental model, I can see value in having 'STAC API' specify how to implement Collections / Catalogs without search. But I also like that STAC Core *can* be implemented by a dynamic server, but that we don't need to specify that.

> Anyway, we can't represent all static STACs in the API it seems...

What does this mean?

Reply:
> Ah, that's interesting on using data rel type instead of child, I'd thought they were child. I do think that's a confusing little area.

It makes sense, because it returns you a different JSON. Not a collection/catalog, but the response of /collections, which contains collections and links. /collections then lists the "children" instead of adding (in our case) 450+ links to the links in the landing page. It's just that this is not specified in STAC itself, but only in API (originating from Features). 

> I think the conformance classes are probably the way to go

Agreed, see #27.

> I can see value in having 'STAC API' specify how to implement Collections / Catalogs without search.

Actually, I'd rather say we should have collection search at some point and then there might be no "without search" anylonger.

> > Anyway, we can't represent all static STACs in the API it seems...
> 
> What does this mean?

If we merge this PR, you need Items in collections/catalogs, otherwise the API is not compliant although it's a valid STAC.

Reply:
Ok, cool. Now I understand. Thanks.

> Actually, I'd rather say we should have collection search at some point and then there might be no "without search" any longer.

I like that. And I think records is actually getting close to a draft, so perhaps we could try to put that in for beta.2.

I think I'll cut this PR down to just be the intro text, and then aim to do #42 with #27. Have a branch started now, will push when it's to a decent point.

Reply:
Catching up, think this all was largely resolved in Monday's discussion.

However, regarding @m-mohr statement about not adding child links to root catalog when you have too many collections....do we have another open discussion on this?  In stac-cmr there are a few providers that have more than 1000 collections, and currently they are all added to the root endpoint, e.g., https://cmr.uat.earthdata.nasa.gov/stac/GES_DISC

Reply:
Hmm, my statement about adding child links to root is a bit biased by openEO and is also weakened by the discussion on Monday. I don't think there's a major issue with your implementation. child links at the landing page are fine, BUT I'd recommend to put them in /collections and link to it with rel type data if OAFeat is implemented anyway. Also, in an API that is not necessarily only STAC (as openEO is), it would bloat the initial request with data that is not required at that stage and thus connection times can slow down. In an API just covering STAC, I'd find it's not a major problem as it's the primary goal/content of the API.

Reply:
I think I'd just remove the whole new chapter for now, merge this and then write-up the conformance classes. Those will basically change the whole chapter anyway.

Reply:
My main interest is in having the API browsable by STAC browser, so I think that will need an update to respect the `data` rel type link.

Reply:
Yes, that was my plan - remove the whole chapter on compliance, and then redo it with the conformance PR.

Reply:
@matthewhanson I implemented the data rel type some days ago into STAC Browser: https://github.com/radiantearth/stac-browser/pull/59 Already live on stacindex.org...